lib/action_view/component/base.rb

```diff
@@ -47,7 +48,8 @@ def render_in(view_context, *args, &block)
         old_current_template = @current_template
         @current_template = self
 
-        @content = view_context.capture(&block) if block_given?
+        @content = view_context.capture(self, &block) if block_given?
```

Comment:
What is the reason for this change?

Reply:
this is what makes the component in scope for the block.